src/Button.js

```diff
@@ -1,18 +1,40 @@
-import React from "react";
-import "./Style.css";
-import constant from "./constants";
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import './calculatorStyle.css';
 
-let colors = constant.colors;
-const Button = props => (
-  <button
-    className="Button"
-    style={{
-      width: props.value === "0" ? "120px" : "60px",
-      backgroundColor: colors[props.index]
-    }}
-  >
-    {props.value}
-  </button>
-);
+class Button extends React.Component {
+    render() {
+        const {
+            value = 'Button',
+            color = 'light gray',
+            width = 'narrow',
+            updateState,
+        } = {
+            ...this.props,
+        };
+        const classes = classNames({
+            'button-style': true,
+            lg: color === 'light gray',
+            dg: color === 'dark gray',
+            og: color === 'orange',
+            nrw: width === 'narrow',
+            wd: width === 'wide',
+        });
+
+        return (
+            <button className={classes} onClick={() => updateState(value)}>
+                {value}
+            </button>
+        );
+    }
+}
+
+Button.propTypes = {
+    value: PropTypes.string.isRequired,
+    color: PropTypes.oneOf(['dark gray', 'orange']),
```

Comment:
Why isn't `light grey` here? By the way, please use `-`, like this: `dark-grey`.